eZ/Publish/Core/FieldType/Image/ImageStorage/Gateway/DoctrineStorage.php

```diff
@@ -456,4 +456,38 @@ public function updateImagePath(int $fieldId, string $oldPath, string $newPath):
             ->execute()
         ;
     }
+
+    /**
+     * @throws \eZ\Publish\Core\IO\Exception\InvalidBinaryFileIdException
+     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws \Doctrine\DBAL\Exception
+     */
+    public function hasImageReference(string $uri, int $fieldId): bool
+    {
+        $path = $this->redecorator->redecorateFromSource($uri);
+
+        $selectQuery = $this->connection->createQueryBuilder();
+        $selectQuery
+            ->select(1)
+            ->from($this->connection->quoteIdentifier(self::IMAGE_FILE_TABLE))
+            ->where(
```

Comment:
`where` `QueryBuilder`'s method should be avoided, because it has a side-effect of discarding any previous where statements that might have been added. I know that is not the case here (there are none), but in general Doctrine recommends using only `andWhere`.
```suggestion
            ->andWhere(
```